harvard-university-for-the-creative-arts.csl

```diff
@@ -167,20 +203,25 @@
           </group>
         </group>
       </else-if>
-      <else-if type="map" match="any">
-        <group delimiter=" ">
-          <text variable="genre" prefix="[" suffix="]"/>
-          <text variable="scale" suffix="."/>
-          <text variable="publisher-place" suffix=":"/>
-          <text variable="publisher" suffix="."/>
-          <text variable="collection-title" prefix="(" suffix=")"/>
-        </group>
-      </else-if>
       <else>
-        <group delimiter=": ">
-          <text variable="publisher-place"/>
-          <text variable="publisher" suffix="."/>
-        </group>
+        <choose>
+          <if match="any" variable="publisher" type="motion_picture broadcast report musical_score song entry-dictionary entry-encyclopedia">
+            <text variable="publisher" suffix="."/>
+          </if>
+          <else>
+            <text value="s.n." prefix="(" suffix=")."/>
+          </else>
+        </choose>
+      </else>
+    </choose>
+  </macro>
+  <macro name="publisher-place">
+    <choose>
+      <if match="any" variable="publisher-place" type="broadcast motion_picture report musical_score song entry-dictionary entry-encyclopedia manuscript">
+        <text variable="publisher-place"/>
+      </if>
```

Comment:
> I only want s.l. to appear on a source where I would expect a publisher location and it doesn't have one.
>
> At the moment the code will put in an s.l. on a book when the publisher field is blank, which is what I want.

Right. But what do you want to happen for e.g. reports, which have the publisher-place field (at least in Zotero: https://aurimasv.github.io/z2csl/typeMap.xml#map-report). With your current code, a report without a publisher-place will print nothing, instead of "(s.l.)", because all report items test positive for the if-statement.